src/server/pfs/server/driver.go

```diff
@@ -1738,24 +1726,11 @@ func (d *driver) inspectBranch(txnCtx *txncontext.TransactionContext, branch *pf
 	return result, nil
 }
 
-func (d *driver) listBranch(ctx context.Context, repo *pfs.Repo, reverse bool, cb func(*pfs.BranchInfo) error) error {
-	// Validate arguments
-	if repo == nil {
-		return errors.New("repo cannot be nil")
-	}
-
-	if err := d.env.AuthServer().CheckRepoIsAuthorized(ctx, repo, auth.Permission_REPO_LIST_BRANCH); err != nil {
+func (d *driver) listBranch(ctx context.Context, reverse bool, cb func(*pfs.BranchInfo) error) error {
+	if _, err := d.env.AuthServer().WhoAmI(ctx, &auth.WhoAmIRequest{}); err != nil && !auth.IsErrNotActivated(err) {
 		return err
-	}
-
-	// Make sure that the repo exists
-	if repo.Name != "" {
-		if err := d.repos.ReadOnly(ctx).Get(repo, &pfs.RepoInfo{}); err != nil {
-			if col.IsErrNotFound(err) {
-				return pfsserver.ErrRepoNotFound{Repo: repo}
-			}
-			return err
-		}
+	} else if err == nil {
+		return errors.New("Cannot list branches from all repos with auth activated")
```

Comment:
this is effectively what happened before, since we were checking for a role binding for `REPO:`